File: Blind150/2.Two Pointer/Python/container_with_most_water.py
# https://leetcode.com/problems/container-with-most-water/description/

from typing import List

class Solution:
    def maxArea(self, heights: List[int]) -> int:

        l, r = 0, len(heights)-1

        max_area = 0

        while l < r:

            cur_area = (r-l)*min(heights[l], heights[r])

            max_area = max(max_area, cur_area)

            if heights[l] < heights[r]:

                l += 1
            else:

                r -= 1
        return max_area

File: Blind150/2.Two Pointer/GO/container_with_most_water.go
// https://leetcode.com/problems/container-with-most-water/description/

func maxArea(height []int) int {

    l,r := 0, len(height) - 1

    maxArea := 0

    for l < r {

        curArea := (r-l)*min(height[l], height[r])

        maxArea = max(maxArea, curArea)

        if height[l] < height[r] {

            l = l + 1
        }else{
            r = r - 1
        }
    }
    return maxArea
}
